src/components/exportAndImport/backup.ts

```diff
@@ -18,46 +35,120 @@ const passwordArg: string | undefined = process.argv[3]
 
 const createBackup = async (password?: string) => {
   // Take snapshot
-  const snapshotName = `${backupFilePrefix}_${DateTime.now().toFormat('yyyy-LL-dd_HH-mm-ss')}`
+  const snapshotName = backupFilePrefix
+  await fsx.ensureDir(path.join(BACKUPS_FOLDER, SNAPSHOT_ARCHIVES_FOLDER_NAME))
 
   console.log(`Creating system backup: ${snapshotName}`)
 
   if (!password) console.log('WARNING: Backup is not encrypted')
 
-  await takeSnapshot({ snapshotName, optionsName: 'backup' })
-
-  // Zip it using password (or unencrypted if no password)
-  const output = fs.createWriteStream(path.join(BACKUPS_FOLDER, `${snapshotName}.zip`))
-  const archive = password
-    ? archiver.create('zip-encrypted', {
-        zlib: { level: 9 },
-        encryptionMethod: 'aes256',
-        password,
-      } as any)
-    : archiver('zip', { zlib: { level: 9 } })
-
-  output.on('close', () => {
-    console.log(`Backup done`)
-    execSync(`rm -rf ${path.join(SNAPSHOT_FOLDER, snapshotName)}`)
-    cleanUpBackups().then(() => {
-      if (isManualBackup) process.exit(0)
+  let backupInfo: BackupInfo
+  try {
+    backupInfo = await fsx.readJSON(path.join(BACKUPS_FOLDER, 'backup.json'), { throws: false })
+  } catch {
+    backupInfo = { latestBackup: null, archives: [] }
+  }
+
+  // Get ids of all existing archive backups (and confirm snapshots still exist)
+  const existingArchiveBackupIds = await getArchiveBackups(backupInfo)
+
+  // Get ids of all current system archives
+  let currentSystemArchives: ArchiveInfo[]
+  try {
+    const archiveInfo: ArchiveData = await fsx.readJSON(
+      path.join(FILES_FOLDER, ARCHIVE_SUBFOLDER_NAME, 'archive.json')
+    )
+    currentSystemArchives = archiveInfo.history
+  } catch {
+    currentSystemArchives = []
+  }
+
+  // Compare them and collect all missing
+  const archivesNotBackedUp = currentSystemArchives.filter(
+    (archive) => !existingArchiveBackupIds.includes(archive.uid)
+  )
+
+  const zipSources: string[] = []
+
+  // Make new archive snapshot with missing archives
+  if (archivesNotBackedUp.length > 0) {
+    const archiveFrom = Math.min(...archivesNotBackedUp.map((a) => a.timestamp))
+    const archiveTo = Math.max(...archivesNotBackedUp.map((a) => a.timestamp))
```

Comment:
Strictly speaking, this doesn't just get the specific archives found, it collects a whole range between the earliest and latest. But in most normal operation, this will be the same thing.